apps/service-catalog/middleware.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  hasOrganizationReadPermission,
+  hasOrganizationWritePermission,
+  validOrganizationNumber,
+  validUUID,
+} from '@catalog-frontend/utils';
+import { NextResponse } from 'next/server';
+import { NextRequestWithAuth, withAuth } from 'next-auth/middleware';
+import { Service } from '@catalog-frontend/types';
+import { handleGetPublicServiceById, handleGetServiceById } from '@catalog-frontend/data-access';
+
+export const config = {
+  matcher: '/catalogs/:path*',
+};
+
+export default withAuth(async function middleware(req: NextRequestWithAuth) {
+  const validType = (type: string): boolean => {
+    if (type === 'services' || type === 'public-services') {
+      return true;
+    }
+    return false;
+  };
+
+  const accessToken = req.nextauth.token?.access_token;
+
+  const pathname = req.nextUrl.pathname;
+  const parts = pathname.split('/');
+  const catalogId = validOrganizationNumber(parts[2]) ? parts[2] : '';
+  const type = validType(parts[3]) ? parts[3] : undefined;
+  const serviceId = validUUID(parts[4]) ? parts[4] : undefined;
+
+  const writePermissionsRoutes = [
```

Comment:
Er dette alle mulige sider? Alt annet blir no-access? Bør kanskje skille mellom not found og no-access.

Reply:
Jeg har allerede denne:

`export const config = {
  matcher: '/catalogs/:path*',
};
`